cmd/detectExecuteScan.go

```diff
@@ -253,9 +254,24 @@ func exitCodeMapping(exitCodeKey int) string {
 
 func getDetectScript(config detectExecuteScanOptions, utils detectUtils) error {
 	if config.ScanOnChanges {
+		log.Entry().Infof("Using Detect Rescan script")
 		return utils.DownloadFile("https://raw.githubusercontent.com/blackducksoftware/detect_rescan/master/detect_rescan.sh", "detect.sh", nil, nil)
 	}
-	return utils.DownloadFile("https://detect.synopsys.com/detect.sh", "detect.sh", nil, nil)
+	if piperutils.ContainsStringPart(config.CustomEnvironmentVariables, "DETECT_LATEST_RELEASE_VERSION") {
+		releaseVersion := ""
+		for _, i := range config.CustomEnvironmentVariables {
+			if strings.Contains(i, "DETECT_LATEST_RELEASE_VERSION") {
+				releaseVersion = strings.Split(i, "=")[1]
+			}
+		}
+		log.Entry().Infof("Using detect script Version %v ", releaseVersion)
+		detect6, _ := regexp.MatchString("6\\.\\d\\.\\d", releaseVersion)
+		if detect6 {
+			log.Entry().Infof("Using Detect 6.x")
+			return utils.DownloadFile("https://detect.synopsys.com/detect.sh", "detect.sh", nil, nil)
+		}
+	}
+	return utils.DownloadFile("https://detect.synopsys.com/detect7.sh", "detect.sh", nil, nil)
```

Comment:
if `config.CustomEnvironmentVariables` is not matching the regex because of typos or an arbitrary string we are always going as a fallback to the latest detect script, right?
Would it be good to at least return an info to the user, that the regex didn't match?

Reply:
I have added an info message which says detect7 is being downloaded, in case of fall back. I hope that should be enough to indicate that the environment variable was incorrectly set, is at all it is set.